doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,256 @@
+plt.close()
+
+"""
+.. figure:: ivim_data_slice.png
+   :align: center
+
+   Heat map of a slice of data
+
+The region around the intersection of the cross-hairs in the figure
+contains cerebral spinal fluid (CSF), so it so it should have a very high $\mathbf{f}$
+and $\mathbf{D^*}$, the area between the right and left is white matter so that
+should be lower, and the region on the right is gray matter and CSF. That should
+give us some contrast to see the values varying across the regions.
+"""
+
+x1, x2 = 160, 180
+y1, y2 = 90, 110
+data_slice = data[x1:x2, y1:y2, z, :]
+
+plt.imshow(data[x1:x2, y1:y2, z, b].T, origin='lower', cmap="gray", interpolation=None)
+plt.savefig("CSF_slice.png")
+plt.close()
+
+"""
+.. figure:: CSF_slice.png
+   :align: center
+
+   Heat map of the CSF slice selected.
+
+Now that we have prepared the datasets we can go forward with
+the ivim fit. Instead of fitting the entire volume, we focus on a
+small section of the slice as selected aboove, to fit the IVIM model.
+First, we instantiate the Ivim model. Using a two-stage approach: first,
+a tensor is fit to the data, and then initial guesses for the parameters
+$\mathbf{S_{0}}$ and $\mathbf{D}$ obtained from this this tensor by
+``_estimate_S0_D`` is used as the starting point for the non-linear fit
+of IVIM parameters using Scipy's ``leastsq`` or ``least_square`` function
+depending on which Scipy version you are using. All initializations for
+the model such as ``split_b`` are passed while creating the ``IvimModel``.
+If you are using Scipy 0.17, you can also set bounds by setting
+``bounds=([0., 0., 0., 0.], [np.inf, 1., 1., 1.]))`` while initializing
+the IvimModel. It is recommeded that you upgrade to Scipy 0.17 since
+the fitting results might at times return values which do not make sense
+physically. (For example a negative $\mathbf{f}$)
+"""
+
+ivimmodel = IvimModel(gtab)
+
+"""
+To fit the model, call the `fit` method and pass the data for fitting.
+"""
+
+ivimfit = ivimmodel.fit(data_slice)
+
+"""
+The fit method creates a IvimFit object which contains the
+parameters of the model obtained after fitting. These are accessible
+through the `model_params` attribute of the IvimFit object.
+The parameters are arranged as a 4D array, corresponding to the spatial
+dimensions of the data, and the last dimension (of length 4)
+corresponding to the model parameters according to the following
+order : $\mathbf{S_{0}, f, D^*, D}$.
+"""
+
+ivimparams = ivimfit.model_params
+print("ivimparams.shape : {}".format(ivimparams.shape))
+
+"""
+As we see, we have a 20x20 slice at the height z = 27. Thus we
+have 400 voxels. We will now plot the parameters obtained from the
+fit for a voxel and also various maps for the entire slice.
+This will give us an idea about the diffusion and perfusion in
+that section. Let(i, j) denote the coordinate of the voxel. We have
+already fixed the z component as 27 and hence we will get a slice
+which is 33 units avove.
```

Comment:
I am not sure I understand where the number 33 comes from. Also, "avove" => "above"